Source/Charts/Data/Implementations/Standard/ChartDataSet.swift

```diff
@@ -217,7 +217,7 @@ open class ChartDataSet: ChartBaseDataSet
         rounding: ChartDataSetRounding) -> Int
     {
         var closest = partitioningIndex { $0.x >= xValue }
-        guard closest < endIndex else { return rounding == .closest ? (endIndex-1) : -1 }
+        guard closest < endIndex else { return index(before: endIndex) }
```

Comment:
Hmm this doesn’t respect the rounding mode though. For example, if `rounding == .up`, there should be no match, right?

Reply:
I'm sure the rounding should not be applied when we are looking for an x higher than the highest in the set. 

The rounding is used for choosing between two x's when in range. Looking for x = 1.0 in [0, 1.5, 3, 4.5, 6, 7.5 ] should return index 0 when down, index 1 when up and index 1 for closest. Looking for 9.9 should return index 5

Returning a '-1' for both up or down when we hit the last in the range seems invalid.